cibyl/sources/elasticsearch/api.py

```diff
@@ -46,24 +46,31 @@ def __init__(self, driver: str = 'elasticsearch',
                          enabled=enabled)
         self.url = url
         self.es_client = elastic_client
+        try:
+            url_parsed = urlsplit(self.url)
+            self.host = f"{url_parsed.scheme}://{url_parsed.hostname}"
+            self.port = url_parsed.port
+        except ValueError as exception:
+            raise ElasticSearchError(
+                'The URL given is not valid'
+            ) from exception
 
-    def setup(self):
+    def setup(self) -> None:
         """ Ensure that a connection to the elasticsearch server can be
         established.
         """
         if self.es_client is None:
-            try:
-                url_parsed = urlsplit(self.url)
-                host = f"{url_parsed.scheme}://{url_parsed.hostname}"
-                port = url_parsed.port
-            except Exception as exception:
-                raise ElasticSearchError(
-                    'The URL given is not valid'
-                ) from exception
-            self.es_client = ElasticSearchClient(host, port).connect()
+            self.es_client = ElasticSearchClient(
+                self.host,
+                self.port
+            ).connect()
+
+    def teardown(self) -> None:
+        if self.es_client:
+            self.es_client.disconnect()
 
     @speed_index({'base': 1})
-    def get_jobs(self: object, **kwargs: Argument) -> list:
+    def get_jobs(self, **kwargs: Argument) -> list:
```

Comment:
Whenever you type hint a collection, try to indicate what it contains. You can do it like this: 'List[str]'.

Reply:
I merged main and it's already changed. Thanks.